pkg/syncer/spec/spec_controller.go

```diff
@@ -168,10 +169,14 @@ func NewSpecSyncer(syncerLogger logr.Logger, syncTargetWorkspace logicalcluster.
 					if namespace != "" {
 						// Use namespace lister
 						nsObj, err := namespaceLister.Get(namespace)
-						if err != nil {
+						if err != nil && !errors.IsNotFound(err) {
 							utilruntime.HandleError(err)
 							return
 						}
+						if errors.IsNotFound(err) {
+							return
+						}
```

Comment:
```suggestion
						if errors.IsNotFound(err) {
							return
						}
						if err != nil {
							utilruntime.HandleError(err)
							return
						}
```

Would this be cleaner?

Reply:
Or
```go
if err != nil {
  if !errors.IsNotFound(err) {
    utilruntime.HandleError(err)
  }
  return
}
```

Reply:
Sure.